pipelines/tekton/resources.go

```diff
@@ -134,6 +134,10 @@ func generatePipelineRun(f fn.Function, labels map[string]string) *pplnv1beta1.P
 			envs = append(envs, e.KeyValuePair())
 		}
 		buildEnvs.ArrayVal = envs
+	} else {
+		// need to hack empty BuildEnvs array on Tekton 1.8+
```

Comment:
I think the version here is v0.39.0+
```suggestion
		// need to hack empty BuildEnvs array on Tekton v0.39.0+
```
1.8 is openshift pipelines impacted version

Reply:
good catch!